src/cli_util.rs

```diff
@@ -2028,6 +2028,35 @@ impl ValueParserFactory for RevisionArg {
     }
 }
 
+fn resolve_default_command(config: &config::Config, app: &Command, string_args: &mut Vec<String>) {
+    const PRIORITY_FLAGS: &[&str] = &["help", "--help", "-h", "--version", "-V"];
+
+    let has_priority_flag = string_args
+        .iter()
+        .any(|arg| PRIORITY_FLAGS.contains(&arg.as_str()));
+    if has_priority_flag {
+        return;
+    }
+
+    let app_clone = app
+        .clone()
+        .allow_external_subcommands(true)
+        .ignore_errors(true);
+    let matches = app_clone.try_get_matches_from(string_args.clone()).ok();
+    if matches
+        .as_ref()
+        .map_or(true, |m| m.subcommand_name().is_none())
```

Comment:
What cases is this `map_or(true, ...)` needed for? I'm not sure what errors would not be ignored by `ignore_error()`. I'm just checking to make sure that we produce reasonable error messages when we get this kind of error. Specifically, if the user does `jj log <something that clap doesn't like>`, we should not insert a "log" in there and make the full command line `jj log log <something clap doesn't like>` if that might then end up in an error message.

Reply:
You're right, it's no longer necessary. I had arrived at that structure before figuring out the `ignore_errors()` technique, so it can be simplified a bit. I've pushed that change now and incomplete commands do still show the appropriate errors:

```
$ jj branch create
error: the following required arguments were not provided:
  <NAMES>...

Usage: jj branch create <NAMES>...

For more information, try '--help'.
```

Reply:
I just added a test with the `jj -T show` case as well to make sure that edge case is covered.